### PR TITLE
CMake: Add sections to Final Report

### PR DESCRIPTION
**Pull request by berniev**
Final Report doesn't distinguish system settings

Added ability to have 'sections' to group stuff
Added value function to display options as ON / OFF / value / -undefined-
Added sections System, Config, Libraries
Added a bunch of data to System and Config. Moved some out of Libraries as appropriate.

### CODE REVIEW

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8314">pipeline status</a> for feature branch [PR_8314](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8314). Pipeline [763700975](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/763700975) was triggered at [09593c3](https://github.com/FreeCAD/FreeCAD/pull/8314/commits/09593c3fec0adaaa28882001cc96ce96e4e41a2f). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8314">pipeline status</a> for feature branch [PR_8314](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8314). Pipeline [763712708](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/763712708) was triggered at [9fd7ae9](https://github.com/FreeCAD/FreeCAD/pull/8314/commits/9fd7ae9da8297698c0d23510d1f8a8c9e31886c2). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by berniev** [comment]
Corrected Matplotlib

File: cMake/FreeCAD_Helpers/SetupCoin3D.cmake [hunk 1]
```diff
@@ -35,5 +35,9 @@ macro(SetupCoin3D)
     IF(NOT PIVY_VERSION)
       execute_process (COMMAND ${Python3_EXECUTABLE} -c "import pivy as p; print(p.__version__,end='')" OUTPUT_VARIABLE PIVY_VERSION)
     ENDIF()
+    
+    if(PIVY_VERSION)
```
Comment (chennes) [inline cMake/FreeCAD_Helpers/SetupCoin3D.cmake:39]: Any reason this can't be an `ELSE()` of the above `IF()`?

**Comment by berniev** [comment]
Thanks, thats a noop oops -- leftover larger experiment. Removed.

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8314">pipeline status</a> for feature branch [PR_8314](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8314). Pipeline [767338520](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/767338520) was triggered at [ef18cd8](https://github.com/FreeCAD/FreeCAD/pull/8314/commits/ef18cd803ea003cc8c17a55c156649c621f58cea). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by chennes** [comment]
Sorry for the delay, this fell off my radar while I was traveling. In my testing, I am getting an extra blank entry at the end of the "config" and "system" lists. I also suggest that "Python" makes more sense in System than in Libraries (but I won't die on that hill).

File: cMake/FreeCAD_Helpers/PrintFinalReport.cmake [hunk 2]
```diff
@@ -26,19 +38,77 @@ macro(PrintFinalReport)
         simple(${name} ${out})
     endmacro()
 
-    ################ Report data ##################
+    function(section_begin name)
+        unset(simpleLines PARENT_SCOPE)
+        message(STATUS "\n\n   ==============\n"
+                         "     ${name}\n"
```
Comment (luzpaz) [inline cMake/FreeCAD_Helpers/PrintFinalReport.cmake:44]: Line up double quotes under above double quotes

File: cMake/FreeCAD_Helpers/SetupCoin3D.cmake [hunk 3]
```diff
@@ -35,5 +35,5 @@ macro(SetupCoin3D)
     IF(NOT PIVY_VERSION)
       execute_process (COMMAND ${Python3_EXECUTABLE} -c "import pivy as p; print(p.__version__,end='')" OUTPUT_VARIABLE PIVY_VERSION)
     ENDIF()
-
+
```
Comment (luzpaz) [inline cMake/FreeCAD_Helpers/SetupCoin3D.cmake:38]: Clean up trailing whitespace

Comment (berniev) [reply cMake/FreeCAD_Helpers/SetupCoin3D.cmake:38]: The sooner we get clang-format in CI the better.

**Comment by berniev** [comment]
Re blank line: Someone did notice after all! Fixed. And you may have Python wherever you wish! Done too.

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8314">pipeline status</a> for feature branch [PR_8314](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8314). Pipeline [774521304](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/774521304) was triggered at [59d5d47](https://github.com/FreeCAD/FreeCAD/pull/8314/commits/59d5d472a999aa4c1b4249d69f6780c998d8ccc6). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8314">pipeline status</a> for feature branch [PR_8314](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8314). Pipeline [774535141](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/774535141) was triggered at [f396fea](https://github.com/FreeCAD/FreeCAD/pull/8314/commits/f396fea71d7e22acc6bd6806fb69d9f1f80a4993). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by berniev** [comment]
With reference to #8281, may I comment that this PR is an example of a PR being held up by tiny detail. Recent rules attempt to prevent this, to place the onus on the maintainer to make followup changes, but these are being ignored.

I propose a new guideline as to whether a PR should be merged:
"A PR should be merged, without further ado, if the project would be, on balance, better off with it merged".

Suggestions could be given for followup work, encouraging a spirit of friendly cooperation, and trust.

This more closely aligns with how devs with commit rights evaluate their own work.

**Comment by chennes** [comment]
This discussion is going to get lost here -- I suggest moving it to either a new, more focused Issue, or to the original issue you link to.

**Comment by berniev** [comment]
Thanks Chris!

**Pull request merged**